data.tf

```diff
@@ -0,0 +1,14 @@
+data "aws_caller_identity" "this" {}
+
+data "aws_elb_service_account" "main" {}
+
+data "template_file" "bucket_policy" {
+  template = "${file("${path.module}/bucket_policy.json")}"
```

Comment:
Let's you use data-source to create this policy instead of rendering it from a file.

Reply:
It's possible to create data sourced IAM policies but not with S3 policies. Perhaps you're talking about using this: https://www.terraform.io/docs/providers/aws/r/s3_bucket_policy.html but I don't see the advantage as coupling a policy and a bucket seems to make good sense (one requires the other).

It's possible to move from a json file to locals which I think is preferable and I've done that. Also added the ability to provide the policy as a variable if the operator so chooses.